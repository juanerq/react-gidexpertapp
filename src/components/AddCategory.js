import { useState } from "react"
import PropTypes from 'prop-types'

const AddCategory = ({setCategory}) => {

  const [inputValue, setInputValue] = useState('')

  const handleInputValue = (e) => {
    setInputValue( e.target.value )
  }

  const handleSubmit = (e) => {
    e.preventDefault()

    if (inputValue.trim().length > 2) {
      setCategory( cats => [inputValue, ...cats])
      setInputValue('')
    }
  }

  return (
    <form onSubmit={ handleSubmit }>
      <p>{ inputValue }</p>
      <input 
        type='text'
        value={ inputValue }
        onChange={ handleInputValue }
      />
    </form>
  )
}

AddCategory.propTypes = {
  setCategory: PropTypes.func.isRequired
}

export default AddCategory