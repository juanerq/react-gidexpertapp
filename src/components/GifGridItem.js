import PropTypes from 'prop-types'

export const GifGridItem = ({ title, url }) => {
  return (
    <div className="card animate__animated animate__fadeInUp">
      <div className="container-img">
        <img src={ url } alt={ title }/>
      </div>
      <p>{ title }</p>
    </div>
  )
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
}
