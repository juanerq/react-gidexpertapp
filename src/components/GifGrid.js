import { GifGridItem } from './GifGridItem'
import { useFetchGifs } from "../hooks/useFetchGifs"
import PropTypes from 'prop-types'

export const GifGrid = ({ category }) => {

  const { data:images, loading } = useFetchGifs( category )

  return (
    <>
      <h3 className='animate__animated animate__fadeInDown'>{ category }</h3>
      { loading && <p className='animate__animated animate__fadeInDown'>Loading</p> } 
      <div className='container-cards'>
          {
            images.map(img => (
              <GifGridItem key={img.id} {...img} />
            ))
          }
      </div>  
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}