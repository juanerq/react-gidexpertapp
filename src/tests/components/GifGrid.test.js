import { shallow } from 'enzyme'
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs'
jest.mock('../../hooks/useFetchGifs')


describe('Component <GifGrid/>', () => {
  const category = 'Demon Slayer'

  test('It should display GifGrid correctly', () => {

    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    })

    const wrapper = shallow( <GifGrid category={ category } /> )
    expect( wrapper ).toMatchSnapshot()

  })

  test('Should display items when images are loaded', () => {

    const gifs = [{
      id: 'ABC',
      url: 'https://localhost/cualquier/cosa',
      title: 'Cualquier cosa'
    }]

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    })

    const wrapper = shallow( <GifGrid category={ category } /> )
    
    expect( wrapper.find('p').exists() ).toBe( false )
    expect( wrapper.find('GifGridItem').length ).toBe( gifs.length )
    expect( wrapper ).toMatchSnapshot()

  })

})