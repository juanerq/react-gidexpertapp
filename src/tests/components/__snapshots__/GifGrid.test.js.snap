// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component <GifGrid/> It should display GifGrid correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    Demon Slayer
  </h3>
  <p
    className="animate__animated animate__fadeInDown"
  >
    Loading
  </p>
  <div
    className="container-cards"
  />
</Fragment>
`;

exports[`Component <GifGrid/> Should display items when images are loaded 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeInDown"
  >
    Demon Slayer
  </h3>
  <div
    className="container-cards"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquier/cosa"
    />
  </div>
</Fragment>
`;
